cpp/include/raft/matrix/detail/linewise_op.cuh

```diff
@@ -312,8 +314,12 @@ __global__ void __launch_bounds__(BlockSize)
   typedef Linewise<Type, IdxType, VecBytes, BlockSize> L;
   constexpr uint workSize = L::VecElems * BlockSize;
   uint workOffset         = workSize;
+#if defined(__clang__)
```

Comment:
Yes, I'd prefer to pull this out as well. Perhaps, into something like `RAFT_alignas`, since it may be useful in cuml too?

Reply:
This one is the most tricky since clang doesn't know about this construct at all.
We stumbled upon a similar thing in cudf where clang doesn't allow `__device__ [[nodiscard]] ... method(...)` in classes, even though without the `__device__` it would be perfectly fine.
Do you think it would be acceptable to use `__align(n)__` instead?
I'm not sure if in CUDA that is precisely the same as `alignas` in C++, but it would solve the issue with clang at least.